### ISSUE TITLE
Strange behavior regarding the coverage badge

### ISSUE THREAD

**Issue by gvolpe** [lemurheavy/coveralls-public#599]
Hi, I'm creating this issue because there's another one similar but it's closed and inactive since my last comment (https://github.com/lemurheavy/coveralls-public/issues/460).

To recap, the last build reports a coverage of %95.45 but the badge says unknown.

It seems the problem is that Coveralls is creating a new branch for every commit (using the commit hash for the name) but it's taking the master branch for the badge. After synchronize the branches in Coveralls the branch created before disappears and the master branch is the only one remaining.

If you take a look at my repository the only existent branch is master (https://github.com/gvolpe/light-play-rest-api/branches).

This is my project: https://coveralls.io/github/gvolpe/light-play-rest-api. I'm using Codeship for CI.

I'm using this versions:
addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.0.4")
addSbtPlugin("org.scoverage" % "sbt-coveralls" % "1.0.0")

Thanks,
Gabriel.

**Comment by nickmerwin** [comment]
Hi @gvolpe it looks like your Coveralls integration library isn't setting the branch name correctly at build time -- it's using the commit SHA incorrectly instead.

You may want to open an issue with them directly.

Thanks.

**Comment by gvolpe** [comment]
Hi @nickmerwin,

It's working now using travis-ci. But it doesn't work using Codeship. Both Travis and Codeship execute the 'sbt coveralls' command successfully. The main difference is that Travis has the 'after_success' stage where it's actually running 'sbt coveralls'. in Codeship the command is running after 'sbt coverage test' but it doesn't have an 'after_success' stage.

Getting Coveralls to work it's a bit tricky if you want to use Codeship. I prefer this one instead of Travis because it has Continuous Deployment in addition of CI.

In conclusion, I think it's a problem of integration between Codeship and Coveralls, I don't really know if it's a problem of the integration library.

Cheers,
Gabriel.
